webapp/app/.server/service/authorized-users.ts

```diff
@@ -41,9 +41,16 @@ export async function councilMembers(): Promise<Hex[]> {
 
 export type UserRole = z.infer<typeof UserRole>;
 
+export const validateHandlerAddress = async () => {
+  const exists = await l1Rpc.checkContractCode(upgradeHandlerAddress);
+  if (!exists) {
+    throw new Error(`Upgrade handler contract not found at ${upgradeHandlerAddress}`);
+  }
+};
+
 export async function isUserAuthorized(address: Hex) {
+  await validateHandlerAddress();
```

Comment:
We could do this during startup instead of each time, right? If the address is wrong the app should not even start.

Reply:
Good idea

Reply:
Done